ogr/mock_core.py

```diff
@@ -0,0 +1,76 @@
+import functools
+import logging
+from ogr.abstract import PullRequest, PRComment, PRStatus
+import datetime
+
+logger = logging.getLogger(__name__)
+
+
+def readonly(_func=None, return_value=None, return_function=None, text=""):
```

Comment:
please choose a better name of the argument "_func"; starting with underscore looks rough

Reply:
also please annotate types and provide a docstring

Reply:
I've added some annotation and docs string and hope that it is good enough :-)